src/hyperlight_host/src/hypervisor/mod.rs

```diff
@@ -189,6 +176,87 @@ pub(crate) trait Hypervisor: Debug + Send {
     /// Get InterruptHandle to underlying VM
     fn interrupt_handle(&self) -> Arc<dyn InterruptHandle>;
 
+    /// Get regs
+    #[allow(dead_code)]
+    fn regs(&self) -> Result<CommonRegisters>;
+    /// Set regs
+    #[allow(dead_code)]
+    fn set_regs(&mut self, regs: &CommonRegisters) -> Result<()>;
+    /// Get fpu regs
+    #[allow(dead_code)]
+    fn fpu(&self) -> Result<CommonFpu>;
+    /// Set fpu regs
+    #[allow(dead_code)]
+    fn set_fpu(&mut self, fpu: &CommonFpu) -> Result<()>;
+    /// Get special regs
+    #[allow(dead_code)]
+    fn sregs(&self) -> Result<CommonSpecialRegisters>;
+    /// Set special regs
+    #[allow(dead_code)]
```

Comment:
Are this `allow(dead_code)` really needed?

Reply:
yes. In next PR I will consolidate everything